gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/notifications/scheduler/ClientSecretKafkaNotificationDefinitionFactory.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.management.service.impl.notifications.scheduler;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.gravitee.am.management.service.impl.notifications.KafkaNotifierConfiguration;
+import io.gravitee.am.management.service.impl.notifications.NotificationDefinitionUtils;
+import io.gravitee.am.management.service.impl.notifications.notifiers.NotifierSettings;
+import io.gravitee.am.management.service.impl.notifications.template.TemplateProvider;
+import io.gravitee.am.model.Template;
+import io.gravitee.node.api.notifier.NotificationDefinition;
+import io.reactivex.rxjava3.core.Maybe;
+import lombok.RequiredArgsConstructor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static io.gravitee.am.management.service.impl.notifications.NotificationDefinitionUtils.TYPE_KAFKA_NOTIFIER;
+
+@RequiredArgsConstructor
+public class ClientSecretKafkaNotificationDefinitionFactory implements NotificationDefinitionFactory<ClientSecretNotifierSubject> {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ClientSecretKafkaNotificationDefinitionFactory.class);
+
+    private final ObjectMapper mapper;
+    private final KafkaNotifierConfiguration kafkaNotifierConfiguration;
+    private final NotifierSettings notifierSettings;
+
+    @Override
+    public Maybe<NotificationDefinition> buildNotificationDefinition(ClientSecretNotifierSubject object) {
+        try {
+            final Map<String, Object> data = new NotificationDefinitionUtils.ParametersBuilder()
```

Comment:
discussion: maybe we should add a "notification type" information into the data as there is notification about clientSecret expiry and also certificate expiry but other cases may happen in the future and using kafka, we should provide information to allow filtering/automation on the consumer side